airavata-api/airavata-api-server/src/main/java/org/apache/airavata/api/server/handler/AiravataServerHandler.java

```diff
@@ -1747,16 +1761,24 @@ public ExperimentStatus getExperimentStatus(AuthzToken authzToken, String airava
     @Override
     @SecurityCheck
     public void launchExperiment(AuthzToken authzToken, final String airavataExperimentId, String gatewayId)
-            throws TException {
+            throws AuthorizationException, AiravataSystemException, TException {
         RegistryService.Client regClient = registryClientPool.getResource();
+        SharingRegistryService.Client sharingClient = sharingClientPool.getResource();
         try {
             ExperimentModel experiment = regClient.getExperiment(airavataExperimentId);
+            String userId = authzToken.getClaimsMap().get(Constants.USER_NAME);
+            String entityId = experiment.getExecutionId();
```

Comment:
The execution id is the application interface id. You need to fetch the application interface and get the application module id from it, and then check to see if the user has access to the application module id.

Reply:
Done.